src/lib.rs

```diff
@@ -137,8 +132,17 @@ mod arch {
 
         match workload {
             Workload::Command(c) => {
+                let mut escaped = String::new();
+                for (i, arg) in c.iter().enumerate() {
+                    if i > 0 {
+                        escaped.push(' ');
+                    }
+                    escaped
+                        .push_str(&arg.replace(" ", "\\ "));
+                }
+
                 command.arg("-c");
-                command.args(&c);
+                command.arg(&escaped);
```

Comment:
```suggestion
                command.arg(c.join(" "));
```

I think we can omit all of the above and pass all of c as a single arg.